pallets/asset/src/lib.rs

```diff
@@ -1925,16 +1624,16 @@ impl<T: Trait> Module<T> {
             ticker,
         ) + value;
 
-        // Charge the fee.
-        if let Some(op) = protocol_fee_data {
-            T::ProtocolFee::charge_fee(op)?;
-        }
+        // In transaction because we don't want fee to be charged if advancing fails.
```

Comment:
when can it fail?

Reply:
`advance_update_balances` => `advance_schedules` => `next_ids` can fail due to ID allocation overflow.